Dockerfile

```diff
@@ -1,10 +1,24 @@
-FROM golang:1.12
+FROM golang:1.12-alpine as build
 
 WORKDIR /src
 
+# build-base is required for tests
+RUN apk add --no-cache \
+    build-base \
+    git
+
 COPY go.mod go.sum ./
 RUN go mod download
 
 COPY . .
 RUN GO111MODULE=on go install .
 CMD ["whalebrew"]
+
+FROM alpine
+COPY --from=docker:18.06 /usr/local/bin/docker /usr/local/bin/docker
```

Comment:
Docker 18.06 reached EOL, so better to make this `18.09`. The Docker cli does API version negotiation, so it will be able to talk against older versions of the daemon.

You can make this configurable through  build-arg by adding an "alias" build-stage. Using build-args for versions allows people to override/test different versions without modifying the Dockerfile. Make sure to set a default value so that the Dockerfile still builds without specifying a custom value:


```Dockerfile
# at the top of the Dockerfile (before the first FROM)
ARG GO_VERSION=1.12
ARG DOCKER_VERSION=18.06

# create an alias (for easier use, and to allow usage in `COPY --from`)
FROM docker:${DOCKER_VERSION} AS docker-cli

FROM golang:${GO_VERSION}-alpine AS build
....

COPY --from=docker-cli /usr/local/bin/docker /usr/local/bin/docker
```